legacy-sdk/whirlpool/tests/integration/multi-ix/splash_pool.test.ts

```diff
@@ -54,9 +61,343 @@ describe("splash pool tests", () => {
     };
   });
 
-  // TODO: add u64 related test cases when we update the math logic
+  describe("trades on splash pool", () => {
+    type TestVariation = {
+      figure: string;
+      poolTickSpacing: number;
+      poolInitialTickIndex: number;
+      poolLiquidity: BN;
+      tradeMode: "exactIn" | "exactOut";
+      tradeDirection: "AtoB" | "BtoA";
+      tradeTokenAmount: BN;
+      expectedPartialFill: boolean;
+    };
+
+    const testVariations: TestVariation[] = [
+      // Notation
+      //
+      // l: lower tick index for FullRange
+      // u: upper tick index for FullRange
+      // m: MIN_TICK_INDEX (-443636)
+      // x: MAX_TICK_INDEX (+443636)
+      // *: liquidity (flat distribution)
+      // S: trade start
+      // T: trade end
+      //
+      // Limit
+      //
+      // 2^33 is almost max liquidity to realize single side deposit at tick index MIN_TICK_INDEX or MAX_TICK_INDEX
+      // 2^64 is almost max liquidity to realize 50:50 deposit at tick index 0
+      //
+
+      ////////////////////////////////////////////////////////////////////////////////
+      // ExactIn
+      ////////////////////////////////////////////////////////////////////////////////
+
+      // ExactIn, BtoA, min to ...
+      {
+        figure: "(toB) |-----mS----l**********T*********|********************u-----x-----| (toA)",
+        poolTickSpacing: 32768 + 128,
```

Comment:
why +128?

Reply:
32768 is `2^15`.
It can be used as minimum tickspacing for SplashPool, but I used `32896 (32768 + 128)` as tickspacing for testing because it is used in production (mainnet) and we will not use 32768.

The following "rule" is convenient.

1% --> 128 (concentrated) & 32768 + 128 (splash)
n% --> x (concentrated) & 32768 + x (splash)